### DISCUSSION TITLE
Problem with --no-strict-syntax and --use-pades-lta

### DISCUSSION THREAD

**Discussion by pquL1** [MatthiasValvekens/pyHanko#466] [Q&A] [answered]
Hi, 

I'm not sure if this is intended behavior or a bug: I have a PDF that I cannot sign with a strict syntax check. Adding `--no-strict-syntax` resolves the issue, but not when I also use `--use-pades-lta`, then I get the same error as without the first parameter:

```
Error: Failed to read PDF file in strict mode; rerun with --no-strict-syntax to try again.
Error message: Illegal generation 4294967295 for object ID 0.
```

Is Pades LTA incompatible with non-strict syntax checking? Or is the `--no-strict-syntax` maybe lost on the way to the extra time stamp?

Greetings

**Comment by edispring** [comment, marked as answer]
Hi Matthias,

I think I found the reason.
In the method `PdfTBSDocument.async_finish_signing` calling `post_sign.post_signature_processing` a new `IncrementalPdfFileWriter` is instantiated and there is no possibility to pass `strict=False` in this case. https://github.com/MatthiasValvekens/pyHanko/blob/056e6841d91306fd0d8271e446ebdd8d9fe3d9fe/pyhanko/sign/signers/pdf_signer.py#L2857

The same applies when calling `PdfTBSDocument.async_sign_pdf`.

Is it possible to add the `strict` option to these methods?
@MatthiasValvekens Should I make a PR?

Best,
Edi

**Reply by MatthiasValvekens** [reply]
That does sound like a good suggestion. Be my guest!

**Reply by pquL1** [reply]
I just noticed that the fix has already been released a while ago! Really cool, thanks a lot @edispring and @MatthiasValvekens  👏
